File: week2/MarshalLee/DailyTemperatures.cs
using System;

public class Class1
{
    public int[] DailyTemperatures(int[] temperatures)
    {
        Stack<int> stack = new();
        int[] array = new int[temperatures.Length];
        for (int i = temperatures.Length - 1; i >= 0; i--)
        {
            while (stack.Count > 0 && temperatures[stack.Peek()] <= temperatures[i])
                stack.Pop();
            if (stack.Count > 0)
                array[i] = stack.Peek() - i;
            stack.Push(i);
        }
        return array;
        //Dictionary<int, int> dict = new();
        //int j = 0;
        //for (int i = 0; i <= temperatures.Length - 1; i++)
        //{
        //    dict.Add(temperatures[i], 0);
        //    while (j < i)
        //    {
        //        if (dict.ElementAt(j).Key < temperatures[i])
        //        {
        //            dict[temperatures[j]] = i - j;
        //            j++;
        //        }
        //        else
        //        {
        //            break;
        //        }
        //    }
        //}

        //return dict.Values.ToArray();
    }
}

File: week2/kuzey/DailyTemperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Deque<Integer> stack = new ArrayDeque<>();
        int[] ans = new int[temperatures.length];

        for (int i = 0; i < temperatures.length; i++) {
            while (!stack.isEmpty() && temperatures[i] > temperatures[stack.peek()]) {
                int prevDay = stack.pop();
                ans[prevDay] = i - prevDay;
            }
            stack.push(i);
        }

        return ans;
    }
}